remove callout and add tabs as examples

mirnawong1 opened the pull request:
addressing @joellabes' request to remove old callouts for alias, database, and schema pages. also added example tabs at the beginning to help provide users with more example at the beginning of the doc

Commit:
remove callout and add tabs as examples

mirnawong1 committed:
Merge branch 'current' into mwong-remove-callout

mirnawong1 committed:
Merge branch 'current' into mwong-remove-callout

mirnawong1 committed:
Merge branch 'current' into mwong-remove-callout

joellabes commented on website/docs/reference/resource-configs/alias.md:
```diff
@@ -4,10 +4,47 @@ description: "Read this guide to understand the alias configuration in dbt."
+
+Specify a custom alias for a model in your `dbt_project.yml` file. 
+
+For example, if you have a model that calculates `sales_metric`s and want to give it a more user-friendly alias, you can alias it like this:
```
Metrics might be a confusing term to give here given the other sort of metrics dbt supports

mirnawong1 replied:
changed it to `sales_total`

```
models:
  your_project:
    sales_total:
      +alias: sales_dashboard
```

joellabes commented on website/docs/reference/resource-configs/alias.md:
```diff
@@ -4,10 +4,47 @@ description: "Read this guide to understand the alias configuration in dbt."
 datatype: string
```
These are too far away from edited content to make a suggestion directly, but: 
- what does that `resource_types` array do? Does it configure how the docs render somehow? 
- Let's update the `description` to something like "Aliasing a resource allows it to be created in the database with a custom name, instead of relying on the filename"

mirnawong1 replied:
asked in the slack channel and updated desc

joellabes commented on website/docs/reference/resource-configs/alias.md:
```diff
@@ -4,10 +4,47 @@ description: "Read this guide to understand the alias configuration in dbt."
+
+</File>
+</TabItem>
+</Tabs>
```
It looks like snapshots and tests can also be aliased: 
- [Snapshots](https://github.com/dbt-labs/dbt-core/issues/4469) example
- Tests example (the longer-term options in [this comment](https://github.com/dbt-labs/dbt-core/issues/3626#issuecomment-886694083))

joellabes commented on website/docs/reference/resource-configs/alias.md:
```diff
@@ -4,10 +4,47 @@ description: "Read this guide to understand the alias configuration in dbt."
+</Tabs>
 
 ## Definition
 
```
or test or snapshot

joellabes commented on website/docs/reference/resource-configs/alias.md:
```diff
@@ -25,9 +62,8 @@ To learn more about changing the way that dbt generates a relation's `identifier
 ## Usage
```
Do you think we still need this section based on the examples you've provided above? I think we could remove it cos it's largely repeating itself?

mirnawong1 replied:
i was going to do exactly that but then i wondered how many ppl might have the #usage anchor link? and since we can't redirect anchor links to another link, i stayed conservative 😭

joellabes commented on website/docs/reference/resource-configs/alias.md:
```diff
@@ -25,9 +62,8 @@ To learn more about changing the way that dbt generates a relation's `identifier
-Configure a seed's alias in your `dbt_project.yml` file.
 
-The seed at `seeds/country_codes.csv` will be built as a <Term id="table" /> named `country_mappings`.
+In this example, the seed at `seeds/country_codes.csv` will be built as a <Term id="table" /> named `country_mappings`.
```
Although it is useful to make clear that the seed lives at `seeds/country_codes.csv` so maybe that should migrate up to the example file before deletion

joellabes commented on website/docs/reference/resource-configs/database.md:
```diff
@@ -5,10 +5,47 @@ datatype: string
 description: "Read this guide to understand the database configuration in dbt."
```
```suggestion
description: "Override the default database when dbt creates resources in your data platform."
```

joellabes commented on website/docs/reference/resource-configs/database.md:
```diff
@@ -5,10 +5,47 @@ datatype: string
-:::caution Heads up!
-This is a work in progress document. While this configuration applies to multiple resource types, the documentation has only been written for seeds.
+<Tabs>
+<TabItem value="model" label="Model">
```
Is it possible to put tests into a custom database? I assume that's alongside store-failures behaviour? Inferring this from the resource_types array above.

mirnawong1 replied:
hmm good question! have askd in dbt help to confirm but i also did add an example just incase

joellabes commented on website/docs/reference/resource-configs/schema.md:
```diff
@@ -5,10 +5,66 @@ description: "Schema - Read this in-depth guide to learn about configurations in
 datatype: string
```
description: "Override the default schema when dbt creates resources in your data platform."

joellabes commented on website/docs/reference/resource-configs/schema.md:
```diff
@@ -5,10 +5,66 @@ description: "Schema - Read this in-depth guide to learn about configurations in
 
-:::caution Heads up!
-This is a work in progress document. While this configuration applies to multiple resource types, the documentation has only been written for seeds.
+<Tabs>
```
This one's great 🎉

joellabes reviewed (changes requested):
ty @mirnawong1! Added a few comments and tweaks but this is going to be so much better 😍

mirnawong1 committed:
Update website/docs/reference/resource-configs/database.md

Co-authored-by: Joel Labes <joel.labes@dbtlabs.com>

mirnawong1 committed:
Merge branch 'current' into mwong-remove-callout

mirnawong1 commented on website/docs/reference/resource-configs/alias.md:
```diff
@@ -4,10 +4,47 @@ description: "Read this guide to understand the alias configuration in dbt."
+</File>
+
+This would return `analytics.finance.sales_dashboard` in the database, instead of the default `analytics.finance.sales_metrics`.
+</TabItem>
```
or is it `analytics.finance.unique_order_id_test`? [this](https://docs.getdbt.com/reference/resource-properties/tests#custom-test-name) makes me think it's the _ one?

mirnawong1 replied:
do we add config or is that only for `properties.yml`?

mirnawong1 committed:
Merge branch 'current' into mwong-remove-callout

mirnawong1 commented:
hey @joellabes thank you so much for this! I've made changes to it and added items you suggested. i did ask about tests in custom database to confirm, but i also added an example just in case. also i have a question below for you if you can clarify because i see some do and some don't based on the issues you linked above. thank you!

- can you also use config block for all 4 resource types?

Commit:
joel's feedback

Commit:
Merge branch 'mwong-remove-callout' of https://github.com/dbt-labs/docs.getdbt.com into mwong-remove-callout

joellabes commented on website/docs/reference/resource-configs/alias.md:
```diff
@@ -40,3 +56,68 @@ seeds:
+
+<TabItem value="test" label="Tests">
+
+Configure a test's alias in your `dbt_project.yml` file. 
```
You can't create tests in a dbt_project.yml, this would be in a schema.yml file

mirnawong1 replied:
thank you! wait is schema.yml the same as properties.yml? https://docs.getdbt.com/reference/configs-and-properties#schemayml-files

joellabes replied:
Yep!

joellabes commented on website/docs/reference/resource-configs/alias.md:
````diff
@@ -40,3 +56,68 @@ seeds:
+      target_schema: finance
+      unique_key: id
+      strategy: timestamp
+      updated_at: updated_at
+      +alias: updated_at_id
+```
+
+This would return the name `analytics.finance.updated_at_id` in the database.
````
It would, but that's a confusing example given that it's not related to updated_at. Maybe set the alias to `your_snapshot` or something that is related to the original name?

joellabes replied:
Also the + isn't necessary here

joellabes commented on website/docs/reference/resource-configs/alias.md:
```diff
@@ -40,3 +56,68 @@ seeds:
+      - name: order_id
+        tests:
+          - unique
+            +alias: unique_order_id_test
```
The + isn't necessary here

joellabes commented on website/docs/reference/resource-configs/alias.md:
````diff
@@ -40,3 +56,68 @@ seeds:
+            +alias: unique_order_id_test
+```
+
+This would return the name `analytics.finance.orders_order_id_unique_order_id_test` in the database.
````
Pretty sure this is only when using `--store-failures`

joellabes commented on website/docs/reference/resource-configs/database.md:
````diff
@@ -2,38 +2,85 @@
+      +database: staging
+```
+
+This would result in the generated relation being located in the `staging` database, so the full relation name would be `staging.finance.product_categories_data`.
````
Where did the `_data` suffix come from here?

mirnawong1 replied:
oh fudge! i think i copied the <tabs> over from alias and didnt remove the _data bit

joellabes reviewed (changes requested):
Looking better! A couple of tweaks on the net-new stuff again

joellabes commented:
> * can you also use config block for all 4 resource types?

Yes: 

```
models:
  - name: first_model
    config:
      alias: second_model
    columns:
      - name: id
        tests:
          - unique:
              config:
                alias: custom_test_name_when_storing_failures
```
```
snapshots:
  - name: first_snapshot
    config:
      alias: second_snapshot
```

mirnawong1 committed:
Merge branch 'current' into mwong-remove-callout

mirnawong1 committed:
Merge branch 'current' into mwong-remove-callout

Commit:
fold in joel's comments

Commit:
Merge branch 'mwong-remove-callout' of https://github.com/dbt-labs/docs.getdbt.com into mwong-remove-callout

mirnawong1 reviewed (commented):
thank you @joellabes ! ok added your feedback now - can you re-review when you have a sec?

mirnawong1 commented:
going to enable automerge in case it's all good to go

joellabes committed:
Merge branch 'current' into mwong-remove-callout
